x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/helpers.ts

```diff
@@ -38,3 +39,36 @@ export const showRulesTable = ({
 export const caseInsensitiveSort = (tags: string[]): string[] => {
   return tags.sort((a: string, b: string) => a.toLowerCase().localeCompare(b.toLowerCase())); // Case insensitive
 };
+
+export const getSearchFilters = ({
+  query,
+  searchValue,
+  filterOptions,
+  defaultSearchTerm,
+}: {
+  query: Query | null;
+  searchValue: string;
+  filterOptions: Record<string, string | null>;
+  defaultSearchTerm: string;
+}): Record<string, string | null> => {
+  try {
+    const fieldClauses = query?.ast.getFieldClauses();
+
+    if (fieldClauses != null && fieldClauses.length > 0) {
+      const filtersReduced = fieldClauses.reduce<Record<string, string | null>>(
+        (acc, { field, value }) => {
+          acc[field] = `${value}`;
+
+          return acc;
+        },
+        filterOptions
+      );
+
+      return filtersReduced;
+    }
+
+    return { [defaultSearchTerm]: searchValue };
+  } catch {
+    return { [defaultSearchTerm]: searchValue };
```

Comment:
When does this try catch blow up? Is it because it's a reduce without a default value going on here? ;-)

If so, I rarely use reduce without a default value to avoid the situation altogether. As a maintainer this looks highly suspect if I don't know when/why this try/catch is going to be hit without comments.

I'll look at the unit tests next to be like, "Is there a unit test that expresses the try/catch so I can see if it's needed and why?"

Reply:
So the catch clause I added because of the `query.ast.getFieldClauses()` throws an error if the user attempts to search a term that is not defined in the schema like `not_a_property:something`.

Reply:
But oof good catch on the reduce sans default value there. I'll update that and add a test to express when the try/catch blows.